support-frontend/app/controllers/CreateSubscriptionController.scala

```diff
@@ -269,37 +270,32 @@ class CreateSubscriptionController(
       product: ProductType,
       userEmail: String,
   )(implicit request: CreateRequest, writeable: Writeable[String]): Future[Result] = {
-    result
-      .fold(
-        { error =>
-          logErrorDetailedMessage(s"create failed due to $error")
-          val errResult = error match {
-            case err: RequestValidationError =>
-              // Store the error message in the result.header.reasonPhrase this will allow us to
-              // avoid alerting for disallowed email addresses in LoggingAndAlarmOnFailure
-              Result(
-                header = new ResponseHeader(
-                  status = BAD_REQUEST,
-                  reasonPhrase = Some(err.message),
-                ),
-                body = writeable.toEntity(err.message),
-              )
-            case _: ServerError =>
-              InternalServerError
-          }
-          Future.successful(errResult)
-        },
-        { createSubscriptionResponse =>
-          logDetailedMessage("create succeeded")
-          cookies(product, userEmail)
-            .map(cookies =>
-              Accepted(createSubscriptionResponse.asJson)
-                .withCookies(cookies: _*)
-                .discardingCookies(discardIncompleteCheckoutCookie),
+    result.value.flatMap {
```

Comment:
I refactored this because there was a warning about nested stateful monads. I think it looks clearer this way as well.